### PR TITLE
Add DataDog logging for submission validation errors

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > When a Veteran attempts to submit a form that has validation errors, a generic error alert is displayed and related error data is sent to Sentry, but this data isn't accessible within Datadog. After discussing adding DataDog logging with Steve Albers, we were given the green light to implement DataDog logging. This will enable us to associate DataDog RUM sessions with the log. This PR is an initial attempt at adding logging.
  >
  > Note that the app _must_ initialize DataDog logging within the app (using the app's client token, service, application ID and version prior to making the logging (added to the platform submit controller) work.
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > Used DataDog documentation to extrapolate initialization code
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#72855](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72855)

## Testing done

- _Describe what the old behavior was prior to the change_
  > No DataDog logging
- _Describe the steps required to verify your changes are working as expected_
  > Can only be validated once merged into staging
- _Describe the tests completed and the results_
  > Added a new SubmitController unit test; and unskipped the entire set of tests ([disabled in 3/2023](https://github.com/department-of-veterans-affairs/vets-website/pull/23719))
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Initially only our Decision Review apps (HLR, NOD & Supplemental Claim)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/shared/utils/useBrowserMonitoring.js [hunk 1]
```diff
@@ -43,6 +45,37 @@ const initializeRealUserMonitoring = customRumSettings => {
...
+
+  // default settings
+  site: 'ddog-gov.com',
+  // see src/site/constants/vsp-environments.js for defaults
+  env: environment.vspEnvironment(), // 'production'
+  sessionSampleRate: 100,
+  forwardErrorsToLogs: true,
+  forwardConsoleLogs: ['error'],
+  forwardReports: [],
+  telemetrySampleRate: 20, // default
```
Comment (data-doge) [inline src/applications/appeals/shared/utils/useBrowserMonitoring.js:64]: Is telemetry the thing that provides the error logs we're looking for? And if so, is there any reason why we can't bump it up to 100?

File: src/platform/forms-system/src/js/review/SubmitController.jsx [hunk 2]
```diff
@@ -117,6 +118,21 @@ class SubmitController extends Component {
...
+        window.DD_LOGS.logger.error(
+          'Validation issue not displayed',
+          {
+            errors: processedErrors,
+            rawErrors: errors,
+            inProgressFormId,
+            userId: user.profile.accountUuid,
+          },
+          'validationError',
+          new Error('ValidationError'), // necessary?
```
Comment (data-doge) [inline src/platform/forms-system/src/js/review/SubmitController.jsx:132]: What does this third argument do?

Comment (data-doge) [reply src/platform/forms-system/src/js/review/SubmitController.jsx:132]: And does it have any negative consequences to the UI?

File: src/applications/appeals/shared/utils/useBrowserMonitoring.js (hunk 1 above)
Comment (data-doge) [reply src/applications/appeals/shared/utils/useBrowserMonitoring.js:64]: Looked this up with @Mottie - unclear from docs whether this applies to error logs from our app, or DD's SDK only. Will merge as-is, and just see what happens in staging.

**Review by data-doge — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

**Comment by Mottie** [comment]
This is working great!

**Pull request merged**